src/test/fixtures/profiles/processed-profile.js

```diff
@@ -814,3 +815,64 @@ export function getCounterForThread(
   };
   return counter;
 }
+
+/**
+ * Get a profile with JS allocations. The allocations will form the following call tree.
+ *
+ * - A (total: 15, self: —)
+ *   - B (total: 15, self: —)
+ *     - F.js (total: 12, self: —)
+ *       - G.js (total: 12, self: 5)
+ *         - H.js (total: 7, self: —)
+ *           - I (total: 7, self: 7)
+ *     - C (total: 3, self: —)
+ *       - D (total: 3, self: —)
+ *         - E (total: 3, self: 3)
+ */
+
+export function getProfileWithJsAllocations(): * {
+  // First create a normal sample-based profile.
+  const {
+    profile,
+    funcNamesDictPerThread: [funcNamesDict],
+  } = getProfileFromTextSamples(`
+    A  A     A
+    B  B     B
+    C  F.js  F.js
+    D  G.js  G.js
+    E        H.js
```

Comment:
optional nit: in other code I removed the `.` so that it's easy to destructure later. And it's still marked as JS frames.

Reply:
Ah, nice tip.